support/k8s/k8s-workload-registrar/controllers/pod_controller.go

```diff
@@ -0,0 +1,221 @@
+	}
+
+	err := ctrl.NewControllerManagedBy(config.Mgr).
+		For(&corev1.Pod{}).
+		Complete(r)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// Reconcile creates a new SPIFFE ID when pods are created
+func (r *PodReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	var pod corev1.Pod
+	ctx := context.Background()
+
+	if containsString(r.c.DisabledNamespaces, req.NamespacedName.Namespace) {
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.Get(ctx, req.NamespacedName, &pod); err != nil {
+		if !errors.IsNotFound(err) {
+			r.c.Log.WithError(err).Error("unable to fetch Pod")
+			return ctrl.Result{}, err
+		}
+
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	spiffeIdUri := ""
+	switch r.Mode {
+	case PodReconcilerModeServiceAccount:
+		spiffeIdUri = r.makeID("ns/%s/sa/%s", req.Namespace, pod.Spec.ServiceAccountName)
+	case PodReconcilerModeLabel:
+		if val, ok := pod.GetLabels()[r.Value]; ok {
+			spiffeIdUri = r.makeID("%s", val)
+		} else {
+			// No relevant label
+			return ctrl.Result{}, nil
+		}
+	case PodReconcilerModeAnnotation:
+		if val, ok := pod.GetAnnotations()[r.Value]; ok {
+			spiffeIdUri = r.makeID("%s", val)
+		} else {
+			// No relevant annotation
+			return ctrl.Result{}, nil
+		}
+	}
+
+	spiffeId := &spiffeidv1beta1.SpiffeID{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace:   pod.Namespace,
+			Annotations: make(map[string]string),
+		},
+		Spec: spiffeidv1beta1.SpiffeIDSpec{
+			SpiffeId: spiffeIdUri,
+			DnsNames: make([]string, 0),
+			Selector: spiffeidv1beta1.Selector{
+				PodUid:    pod.GetUID(),
+				Namespace: pod.Namespace,
+			},
+		},
+	}
+
+	// Set pod as owner of new SPIFFE ID
+	err := controllerutil.SetControllerReference(&pod, spiffeId, r.Scheme)
+	if err != nil {
+		r.c.Log.WithFields(logrus.Fields{
+			"SpiffeID.Name": spiffeId.Name,
+		}).WithError(err).Error("Failed to set pod as owner of new SpiffeID")
+		return ctrl.Result{}, err
+	}
+
+	// Set pod name as first DNS name
+	spiffeId.Spec.DnsNames = append(spiffeId.Spec.DnsNames, pod.Name)
+
+	err = r.createSpiffeId(ctx, pod.ObjectMeta.Name, spiffeId)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Add label to pod with name of SPIFFE ID
+	if pod.ObjectMeta.Labels["spiffe.io/spiffeid"] != spiffeId.ObjectMeta.Name {
+		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// Retrieve the latest version of Pod before attempting update
+			// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+			if err := r.Get(ctx, req.NamespacedName, &pod); err != nil {
+				r.c.Log.WithError(err).Error("Failed to get latest version of Pod")
+				return err
+			}
+			pod.ObjectMeta.Labels["spiffe.io/spiffeid"] = spiffeId.ObjectMeta.Name
+
+			err = r.Update(ctx, &pod)
+
+			return err
```

Comment:
This assignment is setting the `err` variable outside of the closure....
```suggestion
			return r.Update(ctx, &pod)
```